### PR TITLE
Copy Chmod

### PR DESCRIPTION
**Pull request by adamgordonbell**
Adding `--chmod=` to COPY command.

Fixes https://github.com/earthly/earthly/issues/1817

### CODE REVIEW

File: earthfile2llb/interpreter.go [hunk 1]
```diff
@@ -716,6 +717,15 @@ func (i *Interpreter) handleCopy(ctx context.Context, cmd spec.Command) error {
 	if err != nil {
 		return i.wrapError(err, cmd.SourceLocation, "failed to expand COPY chown: %v", opts.Chown)
 	}
+	expandedMode, err := i.expandArgs(ctx, opts.Chmod, false, false)
+	if err != nil {
+		return i.wrapError(err, cmd.SourceLocation, "failed to expand COPY chmod: %v", opts.Platform)
+	}
+	mode, err := strconv.ParseUint(expandedMode, 10, 32)
```
Comment (vladaionescu) [inline earthfile2llb/interpreter.go:724]: I think we need base 8 here?

Comment (adamgordonbell) [reply earthfile2llb/interpreter.go]: yep, you were right.

**Review by vladaionescu — APPROVED** [review]
Looks good - but the tests are failing due to a new-line ending or something. You can run `earthly +lint` locally.

**Review by vladaionescu — COMMENTED** [review]
Oh one more thing.. this is a new feature, we should enable it with a feature flag only for now.

You'll need to add a new feature flag here: https://github.com/earthly/earthly/blob/main/features/features.go#L37. (e.g. `use-chmod`) and wire it up correctly. This will only enable the feature on `VERSION --use-chmod 0.6` (and in the future in `VERSION 0.7`).

**Comment by vladaionescu** [comment]
Oh and an entry to the changelog would be much appreciated! We usually label everything that we add as "experimental" to start with.

File: tests/copy.earth [hunk 2]
```diff
@@ -192,3 +193,24 @@ copy-dir-containing-invalid-symlink:
     RUN mkdir /data
     RUN echo hi > /data/hello.txt
     RUN cat /symlinks/symlink | grep -w hi
+
+copy-chmod:
+    RUN echo "no chmod - -rw-r-r-"
+    COPY in/root .
+    RUN echo "./root file-mode=$(stat -c %a ./root)"
+    RUN if [[ "$(stat -c %a ./root)" != "644" ]]; then exit 1; fi
```
[the remark this answers is not in the document]
Comment (adamgordonbell) [reply tests/copy.earth]: way better ...

**Review by alexcb — APPROVED** [review]

**Pull request merged**